pandas/plotting/_core.py

```diff
@@ -2202,14 +2202,25 @@ def boxplot(data, column=None, by=None, ax=None, fontsize=None,
     def _get_colors():
         #  num_colors=3 is required as method maybe_color_bp takes the colors
         #  in positions 0 and 2.
-        return _get_standard_colors(color=kwds.get('color'), num_colors=3)
+        #  if colors not provided, use color 2 for medians and 0 for all else
```

Comment:
Not super familiar with this code but at first glance this seems kind of arbitrary - what's the reason for the selection here?

Reply:
I was trying to preserve previous behaviour for cases where `color` is not passed. If that's not required, then we can definitely change the color scheme generation to be simpler or even remove it entirely (falling back on `matplotlib` defaults)

Reply:
Ah OK I see - thanks for clarifying